File: java/custom/com/nokia/mid/ui/DeviceControl.java
package com.nokia.mid.ui;

public class DeviceControl {
    public static final int KEYMAT_ALPHANUMERIC = 2;
    public static final int KEYMAT_DEFAULT = 0;
    public static final int KEYMAT_NUMERIC = 1;
    public static final int KEYMAT_OFF = 3;

    public static void setLights(int num, int level) {
        System.out.println("DeviceControl::setLights(int,int) not implemented");
    }

    public static void flashLights(long duration) {
        throw new RuntimeException("DeviceControl::flashLights(long) not implemented");
    }

    public static native void startVibra(int freq, long duration);

    public static native void stopVibra();

    public static int getUserInactivityTime() {
        throw new RuntimeException("DeviceControl::getUserInactivityTime() not implemented");
    }

    public static void resetUserInactivityTime() {
        throw new RuntimeException("DeviceControl::resetUserInactivityTime() not implemented");
    }
}

File: midp/device_control.js
/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */

'use strict';

Native.create("com/nokia/mid/ui/DeviceControl.startVibra.(IJ)V", function(freq, longDuration) {
  // If method is called during a previously called vibration that has been
  // activated from this method, the previous vibration is stopped and the new
  // one is activated using the new set of parameters.
  navigator.vibrate(0);

  // Value 0 can be used for detecting whether or not there is a vibration device.
  if (freq === 0) {
    return;
  }

  var duration = longDuration.toNumber();

  if (freq < 0 || freq > 100 || duration < 0) {
    throw new JavaException("java/lang/IllegalArgumentException");
  }

  navigator.vibrate(duration);
});

Native.create("com/nokia/mid/ui/DeviceControl.stopVibra.()V", function() {
  navigator.vibrate(0);
});
